### PR TITLE
TS-4976: Regularize plugins - protocol

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
Incremental work on #1114.

This also updates the documentation on building a protocol plugin. In addition to changes required by the standardization of the plugin code, other errors were correct. Literal references were converted to links as much as possible.

### CODE REVIEW

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/105/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/92/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/89/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1787/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1680/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/218/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/350/

File: example/protocol/Protocol.c [hunk 1]
```diff
@@ -122,33 +122,29 @@ TSPluginInit(int argc, const char *argv[])
   server_port = 4666;
 
   if (argc < 3) {
-    TSDebug("protocol", "Usage: protocol.so accept_port server_port");
+    TSDebug(PLUGIN_NAME, "Usage: protocol.so accept_port server_port");
     printf("[protocol_plugin] Usage: protocol.so accept_port server_port\n");
```
Comment (maskit) [inline example/protocol/Protocol.c]: Should we keep these two?

**Comment by SolidWallOfCode** [comment]
@maskit 's comment reminds me that I also remove a number of `printf` statements which seemed to duplicate the debug output.

File: example/protocol/Protocol.c (hunk 1 above)
Comment (SolidWallOfCode) [inline example/protocol/Protocol.c]: Probably not, I must have missed that one.

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/108/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/95/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/92/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1790/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1683/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/221/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/353/

**Pull request merged**